nexus/deployment/src/blueprint_builder.rs

```diff
@@ -398,7 +398,11 @@ impl<'a> BlueprintBuilder<'a> {
 
                 // Record each of the sled's zones' underlay addresses as
                 // allocated.
-                if let Some(sled_zones) = self.omicron_zones.get(&sled_id) {
+                let sled_zones =
+                    self.omicron_zones.get(&sled_id).or_else(|| {
+                        self.parent_blueprint.omicron_zones.get(&sled_id)
+                    });
+                if let Some(sled_zones) = sled_zones {
```

Comment:
I tried to replace this with the new helper `sled_current_omicron_zones()` below but I get:

```
error[E0502]: cannot borrow `*self` as immutable because it is also borrowed as mutable
   --> nexus/deployment/src/blueprint_builder.rs:377:67
    |
377 |             self.sled_ip_allocators.entry(sled_id).or_insert_with(|| {
    |             -----------------------                -------------- ^^ immutable borrow occurs here
    |             |                                      |
    |             |                                      mutable borrow later used by call
    |             mutable borrow occurs here
...
401 |                 for z in self.sled_current_omicron_zones(sled_id) {
    |                          ---- second borrow occurs due to use of `*self` in closure

For more information about this error, try `rustc --explain E0502`.
```

I understand the complaint and I've run into this a lot but I don't know generally how to avoid it.